packages/react-components/components/Button.tsx

```diff
@@ -30,95 +30,95 @@ export interface ButtonProps {
   // Debounce onPress event so that it is called once on trigger and
   // consecutive calls in given period are ignored.
-  debouncedOnPress = debounce(this.props.onPress, BUTTON_TAP_DEBOUNCE_TIME, DEBOUNCE_OPTIONS)
+  const debouncedOnPress = useCallback(
+    debounce(props.onPress, BUTTON_TAP_DEBOUNCE_TIME, DEBOUNCE_OPTIONS),
+    [props.onPress]
+  )
 
-  componentDidUpdate(prevProps: ButtonProps, prevState: ButtonProps) {
-    if (this.props.onPress !== prevProps.onPress) {
-      this.debouncedOnPress.cancel()
-      this.debouncedOnPress = debounce(
-        this.props.onPress,
-        BUTTON_TAP_DEBOUNCE_TIME,
-        DEBOUNCE_OPTIONS
-      )
-    }
-  }
-
-  render() {
-    const { text, accessibilityLabel, lineHeight, type, disabled, standard, children } = this.props
-    let textColor
-    let backgroundColor
-    let borderColor
+  const {
+    testID,
+    style,
+    text,
+    accessibilityLabel,
+    lineHeight,
+    type,
+    disabled,
+    standard,
+    children,
+  } = props
+  let textColor
+  let backgroundColor
+  let borderColor
 
-    const isPrimary = type === BtnTypes.PRIMARY
-    const isSecondary = type === BtnTypes.SECONDARY
-    const isTertiary = type === BtnTypes.TERTIARY
+  const isPrimary = type === BtnTypes.PRIMARY
+  const isSecondary = type === BtnTypes.SECONDARY
+  const isTertiary = type === BtnTypes.TERTIARY
 
-    switch (type) {
-      case BtnTypes.PRIMARY:
-        textColor = colors.white
-        backgroundColor = disabled ? colors.celoGreenInactive : colors.celoGreen
-        borderColor = disabled ? colors.celoGreenInactive : colors.celoGreen
-        break
-      case BtnTypes.SECONDARY:
-        textColor = disabled ? colors.celoGreenInactive : colors.celoGreen
-        backgroundColor = 'transparent'
-        borderColor = disabled ? colors.celoGreenInactive : colors.celoGreen
-        break
-      case BtnTypes.TERTIARY:
-        textColor = disabled ? colors.inactiveDark : colors.dark
-        backgroundColor = 'transparent'
-        borderColor = 'transparent'
-        break
-      default:
-        if (__DEV__) {
-          throw new Error('No Button Type Specified')
-        }
-        textColor = colors.white
-        backgroundColor = disabled ? colors.celoGreenInactive : colors.celoGreen
-    }
+  switch (type) {
+    case BtnTypes.PRIMARY:
+      textColor = colors.white
+      backgroundColor = disabled ? colors.celoGreenInactive : colors.celoGreen
+      borderColor = disabled ? colors.celoGreenInactive : colors.celoGreen
+      break
+    case BtnTypes.SECONDARY:
+      textColor = disabled ? colors.celoGreenInactive : colors.celoGreen
+      backgroundColor = 'transparent'
+      borderColor = disabled ? colors.celoGreenInactive : colors.celoGreen
+      break
+    case BtnTypes.TERTIARY:
+      textColor = disabled ? colors.inactiveDark : colors.dark
+      backgroundColor = 'transparent'
+      borderColor = 'transparent'
+      break
+    default:
+      if (__DEV__) {
```

Comment:
I would just throw, even in not dev mode. Helps us catch the error in the wild if this slips through cause it would make its way to Sentry

Reply:
Because we provide default values, I think there is a chance it has been used already somewhere without a type. Hence, if we would want to make it strict - I would check all the usage of `Button` first. But this is a bit outside of the scope of this PR. WDYT?